health-checker/src/main/java/io/stargate/health/CheckerResource.java

```diff
@@ -33,12 +45,40 @@ public Response checkLiveness() {
 
   @GET
   @Path("/readiness")
-  public Response checkReadiness() {
-    // Readiness requires a more heavy weight check to decide if the app is able to successfully
-    // respond to traffic
-    if (bundleService.checkIsReady()) {
+  public Response checkReadiness(@QueryParam("check") Set<String> checks) {
+    boolean checkAll = checks == null || checks.isEmpty();
+
+    HashSet<String> requiredChecks =
+        new HashSet<>(checkAll ? bundleService.defaultHeathCheckNames() : checks);
+
+    SortedMap<String, Result> status =
+        healthCheckRegistry.runHealthChecks(
+            (name, healthCheck) -> checkAll || requiredChecks.contains(name));
+
+    boolean started = bundleService.checkBundleStates();
+    boolean dataStoreAvailable = bundleService.checkDataStoreAvailable();
+    boolean ready = started && dataStoreAvailable;
+
+    for (Entry<String, Result> e : status.entrySet()) {
+      String name = e.getKey();
+      requiredChecks.remove(name);
+      boolean healthy = e.getValue().isHealthy();
+
+      if (!healthy) {
+        logger.warn("Failed health check: {}", name);
+        ready = false;
```

Comment:
what will happen if the first health check is not ready, but the second one is? 
You are doing:
```
requiredChecks.remove(name);
```
before checking if it is `healthy`.
As far as I see, this logic will not work as expected if it is not healthy but known.

Reply:
We remove the name from `requiredChecks` to indicate that we processes that checker. If anything is unhealthy then `ready` will be `false` and the HTTP response will be `SERVICE_UNAVAILABLE`